File: backend/Models/Art.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("Art")]
    public class Art
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Lot {  get; set; } = string.Empty;
        [Required]
        public string Title { get; set; } = string.Empty;
        [Required]
        public string Image {  get; set; } = string.Empty;
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal CurrentMarketPrice {  get; set; }
        [Required]
        public string Condition { get; set; } = string.Empty;
        [Required]
        public bool isFramed { get; set; }
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Height { get; set; }
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Width { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        [Required]
        [ForeignKey("Store")]
        public int StoreId { get; set; }
        public Store Store { get; set; }
        [Required]
        [ForeignKey("Category")]
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public Auction Auction { get; set; }
    }
}

File: frontend/src/models/Art.ts
export type ArtPost = {
  Title: string;
  Image: string;
  CurrentMarketPrice: number;
  Condition: string;
  isFramed: boolean;
  Height: number;
  Width: number;
  StyleId: number;
  MediumId: number;
};

export type Art = {
  id: number;
  lot: string;
  title: string;
  image: string;
  currentMarketPrice: number;
  condition: string;
  isFramed: boolean;
  height: number;
  width: number;
  categoryId: number;
  createdBy: number;
};

export type ArtGet = {
  arts: Art[];
  totalCount: number;
};
